pvlib/iotools/panond.py

```diff
@@ -0,0 +1,197 @@
+"""
+Get .PAN or .OND file data into a nested dictionary.
+"""
+
+import io
+
+
+def num_type(value):
+    """
+    Determine if a value is float, int or a string
+    """
+    if '.' in value:
+        try:  # Detect float
+            value_out = float(value)
+            return value_out
+
+        except ValueError:  # Otherwise leave as string
+            value_out = value
+            return value_out
+
+    else:
+
+        try:  # Detect int
+            value_out = int(value)
+            return value_out
+
+        except ValueError:  # Otherwise leave as string
+            value_out = value
+            return value_out
+
+
+def element_type(element):
+    """
+    Determine if an element is a list then pass to num_type()
+    """
+    if ',' in element:  # Detect a list.
+        # .pan/.ond don't use ',' to indicate 1000. If that changes,
+        # a new method of list detection needs to be found.
+        values = element.split(',')
+        element_out = []
+        for val in values:  # Determine datatype of each value
+            element_out.append(num_type(val))
+
+        return element_out
+
+    else:
+        return num_type(element)
+
+
+def parse_panond(fbuf):
+    """
+    Parse a .pan or .ond text file into a nested dictionary.
+
+    Parameters
+    ----------
+    fbuf : File-like object
+        Buffer of a .pan or .ond file
+
+    Returns
+    -------
+    comp : Nested Dictionary
+        Contents of the .pan or .ond file following the indentation of the
+        file. The value of datatypes are assumed during reading. The value
+        units are the default used by PVsyst.
+
+    Raises
+    ------
+
+    Notes
+    -----
+    The parser is intended for use with .pan and .ond files that were created
+    for use by PVsyst. At time of publication, no documentation for these
+    files was available. So, this parser is based on inferred logic, rather
+    than anything specified by PVsyst.
+
+    The parser assumes that the file being parsed uses indendation of two
+    spaces ('  ') to create new level in a nested dictionary, and that
+    key/values pairs of interest are separated using '='. This further means
+    that lines not containing '=' were omitted from the final returned
+    dictionary.
+
+    Additionally, the indented lines often contain values themselves. This
+    leads to a conflict with the .pan/.ond file and the ability of nested a
+    dictionary to capture that information. The solution implemented here is
+    to repeat that key to the new nested dictionary within that new level.
+    Example below.
+
+    Sample file:
+
+    'level1 = first level
+    key1 = value1
+      level2 = second level
+      key2 = value2'
+
+    output:
+
+    {
+    level1: first level
+    key1: value1,
+    level2:{
+        level2: second level,
+        key2: value2
+        }
+    }
```

Comment:
I don't think this is what the function actually returns.  Shouldn't it be:

```python
{
'level1': 'first level',
'key1': {
    'key1': 'value1',
    'level2': 'second level',
    'key2': 'value2'
    }
}
```

Reply:
I think we are saying basically the same thing. I am happy to adopt your wording, if you thing it is more clear. However, I would like to keep 'levelX' be a key that indicated a nest level of the dictionary. Proposed hybrid of our methods:

{
'key1: 'value1'
'key2': 'value2'
'key3': {
    'key3': 'value3'
    'key4': 'value4'
    }
}

Reply:
I updated the example input to match. I am new to github commenting, I hope the below code snippet shows the update I made.

`
Sample file:

    'level1 = first level
    key1 = value1
    key2 = value2
        level2 = second level
        key3 = value3
        key4 = value4
    key5 = value5'

    output:

    level1:{
        level1: first level
        key1: value1,
        key2: value2,
        level2:{
            level2: second level,
            key3: value3,
            key4: value4
            },
        key5: value5
        }

`